components/gitpod-protocol/src/public-api-converter.ts

```diff
@@ -375,4 +383,49 @@ export class PublicAPIConverter {
         result.defaultWorkspaceImage = settings.defaultWorkspaceImage || undefined;
         return result;
     }
+
+    toConfiguration(project: Project): Configuration {
+        const result = new Configuration();
+        result.id = project.id;
+        result.organizationId = project.teamId;
+        result.name = project.name;
+        result.cloneUrl = project.cloneUrl;
+        result.workspaceSettings = this.toWorkspaceSettings(project.settings?.workspaceClasses?.regular);
+        result.prebuildSettings = this.toPrebuildSettings(project.settings?.prebuilds);
+        return result;
+    }
+
+    toPrebuildSettings(prebuilds: import("./teams-projects-protocol").PrebuildSettings | undefined): PrebuildSettings {
+        const result = new PrebuildSettings();
+        if (prebuilds) {
+            result.enabled = prebuilds.enable;
+            result.branchMatchingPattern = prebuilds.branchMatchingPattern;
+            result.branchStrategy = this.toBranchMatchingStrategy(prebuilds.branchStrategy);
+            result.prebuildInterval = prebuilds.prebuildInterval;
+            result.workspaceClass = prebuilds.workspaceClass;
+        }
+        return result;
+    }
+
+    toBranchMatchingStrategy(
+        branchStrategy?: PrebuildSettingsProtocol.BranchStrategy,
+    ): BranchMatchingStrategy | undefined {
+        switch (branchStrategy) {
+            case "default-branch":
+                return BranchMatchingStrategy.DEFAULT_BRANCH;
+            case "all-branches":
+                return BranchMatchingStrategy.ALL_BRANCHES;
+            case "matched-branches":
+                return BranchMatchingStrategy.MATCHED_BRANCHES;
+        }
+        return undefined;
```

Comment:
BRANCH_MATCHING_STRATEGY_UNSPECIFIED ?

Reply:
I would treat UNSPECIFIED as an error case. But tbh I'm not sure and am a bit unhappy with proto forcing me to always add an UNSPECIFIED enum value.

Reply:
But if we have it then clients has to know what to check for?

My understanding for now:
- if property is not marked as optional in proto then it will be always converted to UNSPECIFIED and clients have to check against it
- if property is marked as optional then it can be omitted


cc @csweichel @easyCZ Could you clarify?